src/libstore/build/substitution-goal.cc

```diff
@@ -110,18 +110,12 @@ void PathSubstitutionGoal::tryNext()
         tryNext();
         return;
     } catch (SubstituterDisabled &) {
-        if (settings.tryFallback) {
-            tryNext();
-            return;
-        }
-        throw;
+        tryNext();
```

Comment:
So I think what we need to do is stash the error here. And then at the end of the function, we re-throw the error instead of returning if nothing is found.

Does it make sense how that brings us in alignment with the spec I wrote?